ecommerce/api.py

```diff
@@ -103,3 +111,184 @@ def make_reference_id(order):
     return (
         f"{_REFERENCE_NUMBER_PREFIX}{settings.CYBERSOURCE_REFERENCE_PREFIX}-{order.id}"
     )
+
+
+def get_valid_coupon_versions(coupons, user, auto_only=False):
```

Comment:
Maybe it would make sense to merge `get_eligible_coupons` with this:

```python
def get_valid_coupon_versions(product, user, auto_only=False, code=None):
.....
        # Get the ids of the latest coupon versions
        product_coupon_query = CouponEligibility.objects.select_related("coupon").filter(
            product=product, coupon__enabled=True
        )
        if code:
            product_coupon_query = product_coupon_query.filter(coupon__coupon_code=code)

        cv_latest = (
            CouponVersion.objects.select_related("coupon")
            .filter(coupon__in=product_coupon_query.values_list('coupon', flat=True))
            .order_by("coupon", "-created_on")
            .distinct("coupon")
            .values_list("pk")
        )
```